src/SOS/Strike/hostcoreclr.cpp

```diff
@@ -1068,21 +1049,38 @@ HRESULT SymbolReader::LoadSymbols(___in IMetaDataImport* pMD, ___in IXCLRDataMod
         return E_FAIL;
     }
 
-    DacpGetModuleData moduleData;
-    hr = moduleData.Request(pModule);
+    ArrayHolder<WCHAR> pModuleName = new WCHAR[MAX_LONGPATH + 1];
+    ULONG32 nameLen = 0;
+    hr = pModule->GetFileName(MAX_LONGPATH, &nameLen, pModuleName);
     if (FAILED(hr))
     {
-        ExtOut("LoadSymbols moduleData.Request FAILED 0x%08x\n", hr);
+        ExtOut("LoadSymbols: IXCLRDataModule->GetFileName FAILED 0x%08x\n", hr);
         return hr;
     }
 
-    ArrayHolder<WCHAR> pModuleName = new WCHAR[MAX_LONGPATH + 1];
-    ULONG32 nameLen = 0;
-    hr = pModule->GetFileName(MAX_LONGPATH, &nameLen, pModuleName);
+    DacpGetModuleData moduleData;
+    hr = moduleData.Request(pModule);
     if (FAILED(hr))
     {
-        ExtOut("LoadSymbols: IXCLRDataModule->GetFileName FAILED 0x%08x\n", hr);
+#ifdef FEATURE_PAL
+        ExtOut("LoadSymbols moduleData.Request FAILED 0x%08x\n", hr);
         return hr;
+#else
+        ULONG64 modBase;
+        hr = GetClrModuleImages(pModule, CLRDATA_MODULE_PE_FILE, &modBase);
```

Comment:
There's some logic here that wasn't present before and I'm not too familiar with it. To confirm, is this trying to get the module info through `GetClrModuleImages` to be able to bet the Windows PDB information? Also, previously the moduleData was never modified, is this populated in the failure case just to be able to do `LoadSymbolsForWindowsPDB` on file layout later?

Reply:
This function was in SOS a long time ago and still is in the desktop source code. It isn't needed for .NET Core anymore because the moduleData.Request is always successful, but it isn't in the desktop DAC so the fallback to GetClrModuleImages.  This function takes a IXCLRDataModule and finds the module base address.